code-challenges.test.js

```diff
@@ -19,10 +20,33 @@ const people = [
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."] 
+
+describe("sentenceMaker", () => {
+  it("returns an array with a sentence about each person with their name capitalized.", () => {
+    expect(sentenceMaker(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// ReferenceError: sentenceMaker is not defined
+// good test
 
 // b) Create the function that makes the test pass.
 
+// pseudo code: declare a function called sentenceMaker
+// input: array of objects
+// output: string
+// process: use .map() with the array, key and symbol as the arguments, 
+// return a string with the information provided in the test variable using string interpolation
+
+// const sentenceMaker = (arr) => {
+//   let firstPart = arr.map((obj => obj.name))
+//   let secondPart = arr.occupation
+//   return firstPart
+// }
+//   console.log(sentenceMaker(people))
+
+// Had a blocker here, I'm able to extract the obj, however it returns the array twice. I was planning on transforming each element into a string (.join)? , and then using .toUpperCase on index zero, then perhaps splicing the strings together, then using string interpolation and transform back into an array. also don't think i had the right arguments initially for my .map() function.
```

Comment:
So when I copy and paste this into my code editor and run it through terminal it is only returning the first and last names of the people in the array. I think that an approach I can walk you through could be mapping (like you have) through the array and then we can declare a variable to hold the word after using a .split method. After that we would be able to target the first letter of the first name giving it the upper case method and then joining them together. Repeating this for last name and then returning a string interpolation of the first variable declared (holding the first name) + the second variable declared (holding the last name) and the rest of the sentence.

Reply:
pseudo code helped guide me in the right direction. thanks !